src/lexer.coffee

```diff
@@ -1252,7 +1252,20 @@ REGEX = /// ^
 REGEX_FLAGS  = /^\w*/
 VALID_FLAGS  = /^(?!.*(.).*\1)[imguy]*$/
 
-HEREGEX      = /// ^(?: [^\\/#] | \\[\s\S] | /(?!//) | \#(?!\{) )* ///
+HEREGEX      = /// ^
+  (?:
+      # Match any character, except those that need special handling below.
+      [^\\/#\s]
+      # Match `\` followed by any character.
+    | \\[\s\S]
+      # Match any `/` except `///`.
+    | /(?!//)
+      # Match `#` which is not part of interpolation, e.g. `#{}`.
+    | \#(?!\{)
+      # Comments consume everything until the end of the line, including `///`.
+    | \s+(?:#(?!\{)[^\n]*)?
```

Comment:
It could be argued that HEREGEX and HEREGEX_OMIT should be consistent. The former uses `[^\n]*` while the latter uses `.*`. What do you think?

Reply:
Yes, It's more consistent. `[^\n]*` had it's purpose when when `[^\n]*\n` pattern was used. But since `\n` is already consumed by `[^\\/#\s]`, we can change pattern for comment to    
`\s+(?:#(?!\{).*)?`.